projection/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -195,6 +197,44 @@ private[projection] class R2dbcOffsetStore(
   private val clearOffsetSql: String =
     s"""DELETE FROM $offsetTable WHERE projection_name = $$1 AND projection_key = $$2"""
 
+  object ReadManagementStateQuery {
+
+    val projectionName: Int = 0
+    val projectionKey: Int = 1
+
+    private def bindParam(i: Int) = "$" + (i + 1)
+
+    val statement: String =
+      s"SELECT paused FROM $managementTable WHERE " +
+      s"projection_name = ${bindParam(projectionName)} and " +
+      s"projection_key = ${bindParam(projectionKey)} "
+
+    /* Map rows retrieved by this query */
+    def mapRow(row: Row) =
+      ManagementState(row.get("paused", classOf[java.lang.Boolean]))
+  }
```

Comment:
By wrapping the statement and binding logic in such an object, I think we can make it easier to deal with the bindings. 

I'm thinking on the statements that has variable bindings. We can encapsulate the numbers and eventual index shifting in such a query object.

Reply:
I find this harder to follow than the original indexes.

Reply:
Are you referring to usage of names like `projectionName` instead of int indices? 
Like `bind(0, "projection name")` is better than `bind(projectName, "projection name")`? 

Or the fact that this is encapsulating the whole query binding requirements in one single place? 

There is one thing that this completes remove and that I find hard to keep track is the fact the index `0` refers to `$1`. Maybe not a big deal for small queries, but when we start to have variable params, index shifting or things like `nextParam()` we are just making the code harder to ourselves.

My main goal here is to experiment with something that could solve #135 without resorting to extra string manipulations (find/replace). Keeping the indices, but in a way that we refer to them by name. 

And of course, each query is different and such an object/class could then encapsulate the specifics of each query (like param shifting and alike).

Reply:
I think there is a misunderstanding. I think it's just a convention (maybe some special parsing in r2dbc-postgres) for using parameter names "$1", "$2", "$3". I don't think those numbers are related to the index numbers 0, 1, 2.

That's why I think we should write the sql without numbers.
```
select * from Foo where a = $ and b = $
```
and then have the simple search-replace utility to construct
```
select * from Foo where a = $1 and b = $2
```

Reply:
I meant the indirection to some the query index constant for simple queries vs just using them in string.

Could we try adding support for named params in prepared statements upstream instead (since R2DBC has API for it) or is that completely up to the database flavour rather than the driver?

Reply:
For the record I just expected I could use that when starting out: 

```
.createStatement("SELECT yadi FROM TABLE WHERE yadi > $theParam")
    .bind("theParam", 42)
```

Reply:
> encapsulating the whole query binding requirements in one single place

When we add support for other databases I think it's these pieces that should be pluggable, not the full Dao classes.
* sql statement
* bind parameters
* mapRow

However, it's a bit early to make that refactoring. Should be done together with adding next database.

Reply:
> Could we try adding support for named params in prepared statements upstream instead (since R2DBC has API for it) or is that completely up to the database flavour rather than the driver?

I'm not sure but I think that is already supported. At least it works with named parameters as `bind("$1", ...)`. That was what I started with but changed to index in #21 because it's a bit more efficient (no name lookups).

Reply:
I see, I thought it simply didn't work, but I probably missed out on the $ in the bind param name.

Reply:
> is that completely up to the database flavour rather than the driver 

They explicit say in their docs that it's not the goal to solve that in the driver. 

_The SQL that is used to create a statement can be parameterized by using vendor-specific bind markers. The portability of SQL statements across R2DBC implementations is not a goal._

I also thought that using $something and then passing "something" to the bind method would work out of the box. I'm quite sure that I also passed '$something' without success. Anyway, I will try again.

> When we add support for other databases I think it's these pieces that should be pluggable

I had that in mind as well, but it's too early for it. 
At some point, extracting the queries and the binding logic will become mandatory.

Reply:
Hmm, I went back to see how we did in projections and we have:

* placeholders using `?` for all dialects
* objects indices so we could easily link a value to some name, instead of a meanless int, eg: `UpdateManagementIndices.PAUSED` 
* surprisingly the indices use base 1

I'm wondering if that's more a R2DBC issue and not really DB specific.

Reply:
> However, it's a bit early to make that refactoring. Should be done together with adding next database. 

I missed that comment. So, yes, I also think it's early. I said on my previous comment. 

My idea was more related to make the binding easy to follow. SInce it's a very simple query, I thought it was a good place to experiment with something without having to touch much code. 

But true that while doing it it became clear to me that we will need to abstract the queries and binding away because of different dialects.